### ISSUE TITLE
Using an aliased type yields a syntax error

### ISSUE THREAD

**Issue by fdupret** [abaplint/abaplint#3580]
```abap
REPORT zfoo.
WRITE 'Hello World'.

INTERFACE lif.
  TYPES: BEGIN OF ty_s_structure,
           field TYPE string,
         END OF ty_s_structure.
ENDINTERFACE.


CLASS lcl DEFINITION.
  PUBLIC SECTION.
    INTERFACES lif.

    ALIASES ty_s_structure FOR lif~ty_s_structure.

    METHODS test.

ENDCLASS.


CLASS lcl IMPLEMENTATION.
  METHOD test.
    DATA ls_structure TYPE ty_s_structure. 

    ls_structure-field = `hello world`. 
  ENDMETHOD.
ENDCLASS.
```
This is syntaxically correct but yields false positives in abaplint.
```
/zfoo.prog.abap[26, 5]: Not a structure, type unknown, target(check_syntax)
/zfoo.prog.abap[24, 10]: Variable "LS_STRUCTURE" contains unknown: TY_S_STRUCTURE not found, lookup(unknown_types)
```

**Comment by fdupret** [comment]
Thanks, much better!
I'm still getting an error when using the alias outside the class:
```abap
REPORT zfoo.
WRITE 'Hello World'.

INTERFACE lif.
  TYPES: BEGIN OF ty_s_structure,
           field TYPE string,
         END OF ty_s_structure.
ENDINTERFACE.


CLASS lcl DEFINITION.
  PUBLIC SECTION.
    INTERFACES lif.

    ALIASES ty_s_structure FOR lif~ty_s_structure.

ENDCLASS.

CLASS lcl IMPLEMENTATION.
ENDCLASS.

DATA ls_structure TYPE lcl=>ty_s_structure. 
```
`/zfoo.prog.abap[22, 6]: Variable "LS_STRUCTURE" contains unknown: TY_S_STRUCTURE not found in class or interface(unknown_types)`
